students/MicahBraun/Lesson 09/mailroom_09.py

```diff
@@ -0,0 +1,420 @@
+# ----------------------------------------------------------------------------------------------------------------------
+# NAME: MICAH BRAUN
+# PROJECT: Lesson 09 -- Object Oriented mailroom.py
+# PURPOSE: To delegate responsibilities of the mailroom.py program and making it more efficient by applying OOP
+# principles to simplify objectives and tasks of program scope.
+#
+# DATE: 07/20/2018
+#
+# DESCRIPTION: Refactored original mailroom.py to 3 classes (UI, DonorSuite, and Donor)
+# all of which have their own constructors and methods. By dividing the tasks into multiple classes it was easier
+# to test for errors and debug the program along the way.
+# ----------------------------------------------------------------------------------------------------------------------
+
+import os
+import datetime
+import sys
+
+# ----------------------------------------------------------------------------------------------------------------------
+#  ********************************************    class Donor    *****************************************************
+# ----------------------------------------------------------------------------------------------------------------------
+
+
+class Donor:
+    """
+    Main module for donor data storage and manipulation, contains:
+    __init__()
+    add_donation()
+    get_avg_donation()
+    get_thank_you()
+    get_thank_you_tofile()
+    get_report_row_header()
+    get_report_row_list()
+    __lt__()
+    """
+
+    def __init__(self, name):
+        """
+        Initializing characteristics to carry through module (e.g. name, donations, avg_donation, and getter)
+        :param name:
+        """
+        self.name = name
+        self.donations = []
+        self.avg_donation = 0
+        self.get_avg_donation()
```

Comment:
it makes more sense to call this function `set_avg_donation` or have it as a `property`